### ISSUE TITLE
undefined function home_path/2

### ISSUE THREAD

**Issue by jobfriender** [phoenixframework/phoenix#2228]
### Environment

* Elixir version (elixir -v): 1.4.2
* Phoenix version (mix deps): phoenix 1.2.3 (Hex package) (mix)
* NodeJS version (node -v): v6.2.1
* NPM version (npm -v): 3.9.3
* Operating system: OSX

### Expected behavior
Warn me that I need to use sudo. Don't give me an error that makes me think I didn't add the resource to my routes when I did add it. Rails bundler gem will escalate sudo privileges when needed. We should copy that.

### Actual behavior
undefined function home_path/2

This error happens when the resource isn't added into the router.ex file.

**Comment by chrismccord** [comment]
What command(s) did you run? What was the output? What didn't get generated that should have due to file perms? We need more information than provided here. Thanks!

**Comment by tarrant** [comment]
I encounter this issue anytime I generate new models:

```
mix phoenix.new test_repo
# Fetch and install dependencies? [Yn] y
cd test_repo
mix ecto.create
mix phoenix.gen.html Post posts body:string word_count:integer
mix phoenix.server
```

Error:
```
Compiling 3 files (.ex)

== Compilation error on file web/controllers/post_controller.ex ==
** (CompileError) web/controllers/post_controller.ex:23: undefined function post_path/2
    (stdlib) lists.erl:1338: :lists.foreach/2
    (stdlib) erl_eval.erl:670: :erl_eval.do_apply/6
    (elixir) lib/kernel/parallel_compiler.ex:117: anonymous fn/4 in Kernel.ParallelCompiler.spawn_compilers/1
```

**Comment by chrismccord** [comment]
@tarrant your issue is you need to follow the post-generator instructions and add the route to your router. Have you missed that step?

**Comment by tarrant** [comment]
Whoops, you are correct noob error/failing to read. It appeared to be the same error so I attempted to help ;)

**Comment by chrismccord** [comment]
No worries! Glad you got things sorted out :)

**Comment by jobfriender** [comment]
I had to run sudo for the migration.

sudo mix ecto.migrate

It wasn't complaining when I ran

mix ecto.migrate

It complained as if I didn't add the route when I had added it.

I just guessed it was the problem. I need to reinstall Elixir under a non root. I installed it with homebrew.

I follow this tutorial to the letter. I was doing a scaffold.
http://nithinbekal.com/posts/elixir-phoenix-crud-app/

I will debug it. A quick fix is probably just making sure an error is thrown if the migration fails because of lack of sudo privileges.

**Comment by chrismccord** [comment]
You must add the route before running mix echo.migrate, as that will try to compile the app in order to run the migrations, so in this case the error message is correct. Can you retry?

**Comment by josevalim** [comment]
Closing this as it seems to be a lack of changing the router file. Please let us know if you have more information.

**Comment by ballPointPenguin** [comment]
Ran into the same thing in 1.3.0. Adding the route solves it (exactly like the instructions tell us).
It does seem a bit curious at first that a route is required for a db migration to work, but now I see that it isn't really the case. The route is required for the application to compile (given the default controller), which in turn is required for the migrations to run.

tl;dr everything is fine 😸
